src/msg/producer/writer/message_writer.go

```diff
@@ -103,40 +113,83 @@ type messageWriterMetrics struct {
 	messageWriteDelay        tally.Timer
 	scanBatchLatency         tally.Timer
 	scanTotalLatency         tally.Timer
+	enqueuedMessages         tally.Counter
+	dequeuedMessages         tally.Counter
+	processedWrite           tally.Counter
+	processedClosed          tally.Counter
+	processedNotReady        tally.Counter
+	processedTTL             tally.Counter
+	processedAck             tally.Counter
+	processedDrop            tally.Counter
+}
+
+func (m messageWriterMetrics) withConsumer(consumer string) messageWriterMetrics {
+	return newMessageWriterMetricsWithConsumer(m.scope, m.opts, consumer)
 }
 
 func newMessageWriterMetrics(
 	scope tally.Scope,
 	opts instrument.TimerOptions,
 ) messageWriterMetrics {
+	return newMessageWriterMetricsWithConsumer(scope, opts, "")
```

Comment:
I’d probably use “none” or “unset” or “unknown” here since empty string can have strange translations down stream implications (some things strip it, others treat it as label not set but then when you query for it makes it hard, can’t use grafana drop down variables, etc).